src/jit/codegenxarch.cpp

```diff
@@ -2651,9 +2638,14 @@ CodeGen::genStoreRegisterReturnInLclVar(GenTreePtr treeNode)
         getStructReturnRegisters(type0, type1, &retReg0, &retReg1);
 
         assert(retReg0 != REG_NA && retReg1 != REG_NA);
-
+        
```

Comment:
We should noway assert that op1 is  GT_CALL, otherwise it will lead to silent bad codegen gen.